src/ui/public/courier/saved_object/saved_object_loader.js

```diff
@@ -39,9 +39,12 @@ export class SavedObjectLoader {
 
   delete(ids) {
     ids = !_.isArray(ids) ? [ids] : ids;
-    return Promise.map(ids, (id) => {
-      return (new this.Class(id)).delete();
-    });
+
+    const promises = [];
+    for (let i = 0; i < ids.length; i++) {
+      promises.push((new this.Class(ids[i])).delete());
```

Comment:
I suggest a native map here instead of imperatively building the array of promises:

```js
const deletions = ids.map(id => {
  const savedObject = new this.Class(id); // just for clarity's sake
  return savedObject.delete();
});

return Promise.all(deletions);
```